### PR TITLE
resolves#47 Decode sent messages in verbose logger

### PR DESCRIPTION
**Pull request by jackunion**
@lugehorsam should take care of https://github.com/heroiclabs/nakama-js/issues/47

### CODE REVIEW

File: packages/nakama-js/web_socket_adapter.ts [hunk 1]
```diff
@@ -145,23 +145,24 @@ export class WebSocketAdapterText implements WebSocketAdapter {
     }
 
     send(msg: any): void {
+        let payload = null;
```
Comment (lugehorsam) [inline packages/nakama-js/web_socket_adapter.ts:148]: I wanted to confirm no functional changes here?

**Review by lugehorsam — APPROVED** [review]

File: packages/nakama-js/web_socket_adapter.ts (hunk 1 above)
Comment (jackunion) [reply packages/nakama-js/web_socket_adapter.ts:148]: None :) Just removed lines of code and made them reusable

**Review by lugehorsam — CHANGES_REQUESTED** [review]
question for you

File: packages/nakama-js/socket.ts [hunk 2]
```diff
@@ -1034,7 +1036,7 @@ export class DefaultSocket implements Socket {
       }
 
       if (this.verbose && window && window.console) {
-        console.log("Sent message: %o", JSON.stringify(untypedMessage));
+        console.log("Sent message: %o", JSON.stringify(atob(untypedMessage)));
```
[the remark this answers is not in the document]
Comment (jackunion) [reply packages/nakama-js/socket.ts]: Sorry, my bad, fixed

File: packages/nakama-js/socket.ts [hunk 3]
```diff
@@ -765,7 +767,7 @@ export class DefaultSocket implements Socket {
 
     this.adapter.onMessage = (message: any) => {
       if (this.verbose && window && window.console) {
-        console.log("Response: %o", JSON.stringify(message));
+        console.log("Response: %o", atob(JSON.stringify(message)));
```
Comment (lugehorsam) [inline packages/nakama-js/socket.ts:770]: This will base64 decode messages that have not been base64 encoded

Comment (jackunion) [reply packages/nakama-js/socket.ts:770]: If I'm reading `web_socket_adapter.ts` right, the encoding happens in there (lines 162 - 166). Would it be better to remove the encoding from there and do it in the send method of `socket.ts`?

Comment (lugehorsam) [reply packages/nakama-js/socket.ts:770]: I'm not sure if that's related to the issue here -- we want to log the base64 decoded version of the match data payload, right? So what if you just base64 decode the match_data or party_data within verbose mode, and log an object with that decoded data?

Comment (jackunion) [reply packages/nakama-js/socket.ts]: @lugehorsam take a look, should be better now

File: packages/nakama-js/socket.ts [hunk 4]
```diff
@@ -765,6 +767,12 @@ export class DefaultSocket implements Socket {
 
     this.adapter.onMessage = (message: any) => {
       if (this.verbose && window && window.console) {
+        if (message.match_data && message.match_data.data) {
+          message.match_data.data = decode(message.match_data.data);
```
Comment (lugehorsam) [inline packages/nakama-js/socket.ts:771]: isn't this decoding the match data twice now? once in the adapter and now once here?

**Comment by lugehorsam** [comment]
I think let's just limit this PR to only code changes within logging, no need to refactor anything else at the moment.

File: packages/nakama-js/socket.ts (hunk 4 above)
Comment (jackunion) [reply packages/nakama-js/socket.ts]: Removed

File: packages/nakama-js/socket.ts [hunk 5]
```diff
@@ -14,6 +14,8 @@
  * limitations under the License.
  */
 
+import { decode, isValid } from 'js-base64'
```
Comment (lugehorsam) [inline packages/nakama-js/socket.ts:17]: i don't think isvalid is used

File: packages/nakama-js/socket.ts [hunk 6]
```diff
@@ -1034,6 +1036,12 @@ export class DefaultSocket implements Socket {
       }
 
       if (this.verbose && window && window.console) {
+        if (untypedMessage.match_data_send) {
+          untypedMessage.match_data_send.data = decode(untypedMessage.match_data_send.data);
```
Comment (lugehorsam) [inline packages/nakama-js/socket.ts:1040]: this is still modifying the message that will actually get sent down the wire, won't it?

Comment (jackunion) [reply packages/nakama-js/socket.ts]: Changed the code to copy `untypedMessage` and log that. Feel free to share any ideas :)

**Comment by lugehorsam** [comment]
Thanks @jackunion

**Pull request merged**